activerecord/test/cases/attribute_methods_test.rb

```diff
@@ -30,6 +31,24 @@ def setup
     ActiveRecord::Base.send(:attribute_method_patterns).concat(@old_matchers)
   end
 
+  test "aliasing `id` attribute allows reading the column value" do
```

Comment:
I think we need more tests in active model, otherwise this behavior is untested over there and anyone using active model without active record isn't covered by the tests.

Reply:
The problem with testing it directly in Active Model is that we can't come up with a reasonable scenario that represents a real application use-case. 
By having `framework_reserved_method?` method explicitly returning `false` we essentially say "nothing is going to change for Active Model" and in order to verify that the only thing we can do is to run existing tests.

It would be important to add Active Model tests when `framework_reserved_method?` for Active Model becomes smarter but for now we can't test much unless we stub `framework_reserved_method?` in the test but that won't be testing a realistic scenario